xblock/runtime.py

```diff
@@ -932,11 +932,13 @@ def get_asides(self, block):
         Arguments:
             block (:class:`.XBlock`): The block to render retrieve asides for.
         """
-        return [
-            self.get_aside_of_type(block, aside_type)
-            for aside_type
-            in self.applicable_aside_types(block)
-        ]
+        return filter(
+            lambda aside: aside.should_apply_to_block(block),
+            [
+                self.get_aside_of_type(block, aside_type)
+                for aside_type in self.applicable_aside_types(block)
+            ]
```

Comment:
Minor nit: since it's in a comprehension anyway, it might read a bit simpler to use an `if` in the comprehension rather than a `filter` outside of it.

Reply:
I need to filter based on the results of `self.get_aside_of_type` though. If I added a condition in this same comprehension, I would need to repeat the call to `self.get_aside_of_type`, right (i.e.: `if self.get_aside_of_type(block, aside_type).should_apply_to_block(block)`. Could use a nested comprehension as well; your call on whether or not that's preferred to the `filter`. Or I might just be missing a blatantly obvious solution here...

Reply:
Ah, you're right, I was misreading it. Okay, still a nit, but it might be clearer if you made an intermediate variable or some comments that describe what that inner generated list of asides represents. Because I think it's not clear to the casual reader what the difference is between `applicable_aside_types` and `should_apply_to_blocks`